system/lib/pthread/library_pthread.c

```diff
@@ -92,7 +92,7 @@ void emscripten_thread_sleep(double msecs) {
   const double minimumTimeSliceToSleep = 0.1;
 
   // main thread may need to run proxied calls, so sleep in very small slices to be responsive.
-  const double maxMsecsSliceToSleep = emscripten_is_main_browser_thread() ? 1 : 100;
+  const double maxMsecsSliceToSleep = emscripten_is_main_runtime_thread() ? 1 : 100;
```

Comment:
This looks good - might be good to change the comment to say `// main runtime thread may need to ...`